libs/x-content/src/main/java/org/elasticsearch/common/xcontent/MediaTypeParser.java

```diff
@@ -65,8 +51,10 @@ public T fromFormat(String format) {
 
     /**
      * parsing media type that follows https://tools.ietf.org/html/rfc7231#section-3.1.1.1
+     *
      * @param headerValue a header value from Accept or Content-Type
      * @return a parsed media-type
+     * //todo pg should this all be maybe based on a regex?
```

Comment:
I suggest that we benchmark this with a JMH microbenchmark (separately) with a variety of different content types and parameters

Reply:
I wrote a benchmark and regex approach is slower. I does cover more (the exact characters allowed by rfc) but not sure we need this. See the implementation:
https://github.com/pgomulka/elasticsearch/blob/compat/benchmark_regex_vs_splits/benchmarks/src/main/java/org/elasticsearch/benchmark/MediaTypeParsingBenchmark.java

https://github.com/pgomulka/elasticsearch/blob/compat/benchmark_regex_vs_splits/libs/x-content/src/main/java/org/elasticsearch/common/xcontent/MediaTypeParserRegex.java

```
Benchmark                                     Mode  Cnt  Score   Error  Units
MediaTypeParsingBenchmark.parseIter           avgt   30  0.801 ± 0.010  us/op
MediaTypeParsingBenchmark.parseIterNoParams   avgt   30  0.200 ± 0.002  us/op
MediaTypeParsingBenchmark.parseRegex          avgt   30  1.639 ± 0.012  us/op
MediaTypeParsingBenchmark.parseRegexNoParams  avgt   30  0.504 ± 0.011  us/op

```